### PR TITLE
Do not sign the tree from the libevent thread.

### PR DESCRIPTION
**Pull request by pphaneuf**
This is a blocking operation, and doing it from the libevent thread is Very Bad (tm).

### CODE REVIEW

**Comment by pphaneuf** [comment]
Wouldn't you like less blocking, @AlCutter? :wink:

File: cpp/server/ct-server.cc [hunk 1]
```diff
@@ -223,19 +235,14 @@ int main(int argc, char* argv[]) {
                                             &log_signer);
   LogLookup<LoggedCertificate> log_lookup(db);
 
-  // Make sure that we have an STH, even if the tree is empty.
   // TODO(pphaneuf): We should be remaining in an "unhealthy state"
   // (either not accepting any requests, or returning some internal
   // server error) until we have an STH to serve. We can sign for now,
   // but we might not be a signer.
-  SignMerkleTree(&tree_signer, &log_lookup);
+  thread signer(&SignMerkleTree, &tree_signer);
```
Comment (AlCutter) [inline cpp/server/ct-server.cc]: Do we want/need a condition_variable/notification here to block until after the first run has completed?

**Comment by AlCutter** [comment]
Looks lovely to me.

File: cpp/server/ct-server.cc (hunk 1 above)
Comment (pphaneuf) [inline cpp/server/ct-server.cc]: No, if we don't want to deadlock again. :smiley:

**Pull request merged**